### PR TITLE
tests: drivers: Verify power management for SPIM (using s2ram)

### PR DESCRIPTION
**Pull request by nordic-bami**
Add benchmark to test SPIM driver cooperation with s2ram.

### CODE REVIEW

File: tests/benchmarks/multicore/idle_spim/prj.conf [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+CONFIG_PM=y
+CONFIG_PM_S2RAM=y
+CONFIG_PM_S2RAM_CUSTOM_MARKING=y
+CONFIG_PM_DEVICE=y
+CONFIG_PM_DEVICE_RUNTIME=y
+CONFIG_POWEROFF=y
+CONFIG_BOOT_BANNER=n
+CONFIG_NRFS_MRAM_SERVICE_ENABLED=n
+CONFIG_CLOCK_CONTROL=n
```
Comment (nordic-piks) [inline tests/benchmarks/multicore/idle_spim/prj.conf:12]: not needed

File: tests/benchmarks/multicore/idle_spim/src/main.c [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	if (!err) {
+		printk("Error: SPI device is not ready, err: %d\n", err);
+		return 0;
+	}
+
+	while (1) {
+		for (int read_index = 0; read_index < SPI_READ_COUNT; read_index++) {
+			err = spi_read_register(CHIP_ID_REGISTER_ADDRESS, &response);
+			if (!err) {
+				printk("SPI read returned error: %d\n", err);
```
Comment (nordic-piks) [inline tests/benchmarks/multicore/idle_spim/src/main.c:55]: return in case of error to break pattern and detect it at current consumption?

File: tests/benchmarks/multicore/idle_spim/remote/prj.conf [hunk 3]
```diff
@@ -0,0 +1,9 @@
+CONFIG_PM=y
+CONFIG_POWEROFF=y
+CONFIG_CONSOLE=n
+CONFIG_UART_CONSOLE=n
+CONFIG_SERIAL=n
+CONFIG_GPIO=n
+CONFIG_BOOT_BANNER=n
+CONFIG_NRFS_MRAM_SERVICE_ENABLED=n
+CONFIG_CLOCK_CONTROL=n
```
Comment (nordic-piks) [inline tests/benchmarks/multicore/idle_spim/remote/prj.conf:9]: not needed

**Review by nordic-piks — APPROVED** [review]

File: tests/benchmarks/multicore/idle_spim/Kconfig.sysbuild [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor ASA
```
Comment (nordic-segl) [inline tests/benchmarks/multicore/idle_spim/Kconfig.sysbuild:2]: 2024

**Review by nordic-pikr — APPROVED** [review]

File: tests/benchmarks/multicore/idle_spim/sysbuild/nrf54h20dk_nrf54h20_cpuppr.conf
Comment (nordic-segl) [inline tests/benchmarks/multicore/idle_spim/sysbuild/nrf54h20dk_nrf54h20_cpuppr.conf:1]: is this file needed?

Comment (nordic-bami) [reply tests/benchmarks/multicore/idle_spim/sysbuild/nrf54h20dk_nrf54h20_cpuppr.conf:1]: To be removed

**Review by nordic-segl — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]
